### ISSUE TITLE
QField Sync failing with QGIS 3.20

### ISSUE THREAD

**Issue by Waldgeist84** [opengisch/QField#2065]
Hello,

I started using QField three months ago with QGIS 3.18.3. Everything worked fine. 

A few days ago, my Linux (Arch-based) upgraded QGIS to 3.20.0. Since the update, packaging works but synchronizing back claims that it works but the field data is not loaded into the project anymore. Same happens with QGIS 3.20.1 on Ubuntu.
QFieldSync Plugin was updated after the upgrade on both machines.

The issue is not project related. It happens for all my projects that worked fine in 3.18.3. The behaviour was verified by installing QGIS 3.18.3 on Windows and reopening the QGIS project, package and sync back. Then everything is back to normal and works as expected. The windows is however just for test purposes only in my case... 

Let me know if I can test something more.

**Comment by Waldgeist84** [comment]
I managed to get some screenshots illustrating the issue: 

QField View:
![IMG_20210803_214545_819](https://user-images.githubusercontent.com/19704256/128078093-272a928b-f891-481a-80e4-8b38594ba9b2.jpg)

QGIS after Synchronization:
![screenshot_gqis](https://user-images.githubusercontent.com/19704256/128078186-49112a6f-dfdd-4f59-b2ac-45413c48d596.png)

However, I can open the qgs file created during packaging in QGIS together with the downloaded gpkg-file and it shows the correct data: 
![screenshot_gqis_qfield_project](https://user-images.githubusercontent.com/19704256/128078434-5878171d-c8e4-4205-8c30-7b384c85650f.png)

Any ideas?

**Comment by nirvn** [comment]
Can you share the original baume dataset in this PR?

**Comment by Waldgeist84** [comment]
Sure. But it is empty at the beginning. The idea is to have a citizen science project where people can report fruit trees in their area... 

[baeume_dataset.zip](https://github.com/opengisch/QField/files/6929072/baeume_dataset.zip)

**Comment by nirvn** [comment]
@Waldgeist84 , can you share the non-exported project file too?

**Comment by Waldgeist84** [comment]
That should give the details of what we try to do with the trees...

[project.zip](https://github.com/opengisch/QField/files/6929380/project.zip)

**Comment by nirvn** [comment]
@Waldgeist84 , if you want to try something, rename your datasets so it doesn't contain special characters like (ä). Keep your filenames to using only A-Z, a-z, and 0-9 characters.

**Comment by Waldgeist84** [comment]
@nirvn  ok, let me check. I had that as a suspicion too. but it worked in the old version (3.18.3). I'll come back to you

**Comment by Waldgeist84** [comment]
@nirvn unfortunately that did not help. Still the same behaviour. I even tried to start with a fresh project without special characters anywhere and only a few fields (value map, date, and text) but still the same result.

**Comment by nirvn** [comment]
@Waldgeist84 , here's an additional test that'd help: 
- export that project using the qfieldsync plugin 
- open the _exported project_, add a few features in the offline point layer
- save that edited offline point layer dataset
- re-open the origin project, and use the qfieldsync plugin to import that exported project which now has a couple of points

Is that working?

**Comment by Waldgeist84** [comment]
@nirvn No, same thing. I see the data in QGIS when opening the exported version I can save it, and it is still in the GeoPackage when reopening it, but it does not end up in the original project upon synchronization. I did not see any errors.

Is there a way to check for console output?

**Comment by nirvn** [comment]
@Waldgeist84 , can you give a QGIS master nightly build a try? There has been some significant improvements to offline layer handling, it might have fixed things for you. I can successfully sync back and forth on master under linux.

**Comment by Waldgeist84** [comment]
@nirvn sorry took me some time to get that installed and test it. However, it still does not work (see screenshot for QGIS version). Could this be some unmet dependency thing? E.g. some library that causes issues? 

Attaching a screenshot of the environment: 
![screenshot_gqis_dev](https://user-images.githubusercontent.com/19704256/128365191-a833d810-44ab-477b-bdb8-792422871ec5.png)

**Comment by aliasatp** [comment]
I realised the same issue. Sync works in 3.18 release and fails in 3.20 one

**Comment by nirvn** [comment]
@Waldgeist84 , @aliasatp , do you still see the issue running the latest point release of QGIS 3.20?

If so, can you provide the qfieldsync version you are running?

**Comment by aliasatp** [comment]
@nirvn, Hi, the issue still remains in 3.20.3 release. What I'm doing is to open my QGIS (3.18 production) project in 3.20.3 release. I save it. I export into QField format running qfieldsync 3.4.4 version. Everything works in QField but when I sync back to QGIS something goes wrong. During the sync process the features collected in QField appear but at the end of the sync process nothing is stored in QGIS (nor in the DB). If I directly open in QGIS the '.qgs' file from the QFiled package evething is ok , but obviously not syncronized

**Comment by Waldgeist84** [comment]
@nirvn Same behaviour here. I also tried with the fresh 3.20 test project I used last time. That shows the exact same behaviour @aliasatp described too.

**Comment by careyshan** [comment]
Hey, I'm also having the same issue with sync on QGIS 3.20. Thanks
